rpc/network/db/rpc_provider_db_test.go

```diff
@@ -0,0 +1,163 @@
+package db_test
+
+import (
+	"database/sql"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/status-im/status-go/appdatabase"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/rpc/network/db"
+	"github.com/status-im/status-go/rpc/network/testutil"
+	"github.com/status-im/status-go/t/helpers"
+)
+
+type RpcProviderPersistenceTestSuite struct {
+	suite.Suite
+	db             *sql.DB
+	cleanup        func()
+	rpcPersistence *db.RpcProvidersPersistence
+}
+
+func (s *RpcProviderPersistenceTestSuite) SetupTest() {
+	testDb, cleanup := setupTestNetworkDB(s.T())
+	s.db = testDb
+	s.cleanup = cleanup
+	s.rpcPersistence = db.NewRpcProvidersPersistence(testDb)
+}
+
+func (s *RpcProviderPersistenceTestSuite) TearDownTest() {
+	if s.cleanup != nil {
+		s.cleanup()
+	}
+}
+
+func setupTestNetworkDB(t *testing.T) (*sql.DB, func()) {
+	testDb, cleanup, err := helpers.SetupTestSQLDB(appdatabase.DbInitializer{}, "rpc-providers-tests")
+	require.NoError(t, err)
+	return testDb, func() { require.NoError(t, cleanup()) }
```

Comment:
Please use `t.Cleanup`, no need to return the callback from here

Reply:
Not exactly, but similar: [Code Smell #6: Cleanup MobileSignalHandler](https://www.notion.so/Code-Smell-6-Cleanup-MobileSignalHandler-67dda79e7e0f4275b105af557d890b8d?pvs=4)